From 4670be25da9904c187309f2332e676a01073177e Mon Sep 17 00:00:00 2001
From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Fri, 12 Jun 2015 16:01:12 +0800
Subject: [PATCH 2/4] MSVC Builds: Add A Pre-configured config.h.win32

This can be used by the Visual Studio projects so that the parts relevant
to Windows can be built, and the symbols can be exported by overriding
_HB_EXTERN with __declspec(dllexport) extern, which is like what GLib is
currently doing.
---
 config.h.win32 | 158 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 158 insertions(+)
 create mode 100644 config.h.win32

diff --git a/config.h.win32 b/config.h.win32
new file mode 100644
index 0000000..d9cc4d1
--- /dev/null
+++ b/config.h.win32
@@ -0,0 +1,158 @@
+/* config.h.in.  Generated from configure.ac by autoheader.  */
+
+/* The normal alignment of `struct{char;}', in bytes. */
+#define ALIGNOF_STRUCT_CHAR__ 1
+
+/* Define to 1 if you have the `atexit' function. */
+#define HAVE_ATEXIT 1
+
+/* Have cairo graphics library */
+#define HAVE_CAIRO 1
+
+/* Have cairo-ft support in cairo graphics library */
+#define HAVE_CAIRO_FT 1
+
+/* Have Core Text backend */
+/* #undef HAVE_CORETEXT */
+
+/* Define to 1 if you have the <dlfcn.h> header file. */
+/* #undef HAVE_DLFCN_H */
+
+/* Have simple TrueType Layout backend */
+#define HAVE_FALLBACK 1
+
+/* Have FreeType 2 library */
+#define HAVE_FREETYPE 1
+
+/* Define to 1 if you have the `getpagesize' function. */
+/* #undef HAVE_GETPAGESIZE */
+
+/* Have glib2 library */
+#define HAVE_GLIB 1
+
+/* Have gobject2 library */
+#define HAVE_GOBJECT 1
+
+/* Have Graphite2 library */
+#define HAVE_GRAPHITE2 1
+
+/* Have ICU library */
+#define HAVE_ICU 1
+
+/* Have Intel __sync_* atomic primitives */
+/* #undef HAVE_INTEL_ATOMIC_PRIMITIVES */
+
+/* Define to 1 if you have the <inttypes.h> header file. */
+#if !defined (_MSC_VER) || (_MSC_VER >= 1800)
+#define HAVE_INTTYPES_H 1
+#endif
+
+/* Define to 1 if you have the `isatty' function. */
+#define HAVE_ISATTY 1
+
+/* Define to 1 if you have the <memory.h> header file. */
+#define HAVE_MEMORY_H 1
+
+/* Define to 1 if you have the `mmap' function. */
+/* #undef HAVE_MMAP */
+
+/* Define to 1 if you have the `mprotect' function. */
+/* #undef HAVE_MPROTECT */
+
+/* Have native OpenType Layout backend */
+#define HAVE_OT 1
+
+/* Have POSIX threads */
+/* #undef HAVE_PTHREAD */
+
+/* Have PTHREAD_PRIO_INHERIT. */
+/* #undef HAVE_PTHREAD_PRIO_INHERIT */
+
+/* Define to 1 if you have the <sched.h> header file. */
+/* #undef HAVE_SCHED_H */
+
+/* Have sched_yield */
+/* #undef HAVE_SCHED_YIELD */
+
+/* Have Solaris __machine_*_barrier and atomic_* operations */
+/* #undef HAVE_SOLARIS_ATOMIC_OPS */
+
+/* Define to 1 if you have the <stdint.h> header file. */
+#if !defined (_MSC_VER) || (_MSC_VER >= 1600)
+#define HAVE_STDINT_H 1
+#endif
+
+/* Define to 1 if you have the <stdlib.h> header file. */
+#define HAVE_STDLIB_H 1
+
+/* Define to 1 if you have the <strings.h> header file. */
+#ifndef _MSC_VER
+#define HAVE_STRINGS_H 1
+#endif
+
+/* Define to 1 if you have the <string.h> header file. */
+#define HAVE_STRING_H 1
+
+/* Define to 1 if you have the `sysconf' function. */
+/* #undef HAVE_SYSCONF */
+
+/* Define to 1 if you have the <sys/mman.h> header file. */
+/* #undef HAVE_SYS_MMAN_H */
+
+/* Define to 1 if you have the <sys/stat.h> header file. */
+#define HAVE_SYS_STAT_H 1
+
+/* Define to 1 if you have the <sys/types.h> header file. */
+#define HAVE_SYS_TYPES_H 1
+
+/* Have UCDN Unicode functions */
+#define HAVE_UCDN 1
+
+/* Have Uniscribe library */
+#define HAVE_UNISCRIBE 1
+
+/* Define to 1 if you have the <unistd.h> header file. */
+#ifndef _MSC_VER
+#define HAVE_UNISTD_H 1
+#endif
+
+/* Define to 1 if you have the <usp10.h> header file. */
+#define HAVE_USP10_H 1
+
+/* Define to 1 if you have the <windows.h> header file. */
+#define HAVE_WINDOWS_H 1
+
+/* Define to the sub-directory where libtool stores uninstalled libraries. */
+#define LT_OBJDIR ".libs/"
+
+/* Define to the address where bug reports for this package should be sent. */
+#define PACKAGE_BUGREPORT "http://bugs.freedesktop.org/enter_bug.cgi?product=harfbuzz"
+
+/* Define to the full name of this package. */
+#define PACKAGE_NAME "Harfbuzz"
+
+/* Define to the full name and version of this package. */
+#define PACKAGE_STRING "Harfbuzz 0.9.40"
+
+/* Define to the one symbol short name of this package. */
+#define PACKAGE_TARNAME "harfbuzz"
+
+/* Define to the home page for this package. */
+#define PACKAGE_URL "http://harfbuzz.org/"
+
+/* Define to the version of this package. */
+#define PACKAGE_VERSION "0.9.40"
+
+/* Define to necessary symbol if this constant uses a non-standard name on
+   your system. */
+/* #undef PTHREAD_CREATE_JOINABLE */
+
+/* Define to 1 if you have the ANSI C header files. */
+#define STDC_HEADERS 1
+
+/* defines how to decorate public symbols while building */
+#ifdef _MSC_VER
+#define _HB_EXTERN __declspec (dllexport) extern
+#else
+#define _HB_EXTERN __attribute__((visibility("default"))) __declspec (dllexport) extern
+#endif
-- 
2.3.5.windows.8

